app/controllers/albums_controller.rb

```diff
@@ -1,2 +1,45 @@
 class AlbumsController < ApplicationController
+  include SharedMethods
+  before_action :get_album, only:[:show, :destroy, :update, :edit, :upvote]
+  def index
+    @albums = Album.all.order(:upvotes).reverse
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def destroy
+    @album.destroy
+    redirect_to albums_path
+  end
+
+  def update
+    @album.update(album_params[:album]) ? (redirect_to album_path(@album)) : (render :edit)
+  end
+
+  def create
+    album = Album.new(album_params[:album])
+    if album.save
+      redirect_to album_path(album.id)
+    else
+      @album = album
+      render :new
+    end
+  end
+
+  def upvote
+    @album.upvotes += 1
```

Comment:
You can write `@album.upvotes.increment!` which will auto-increment the value by 1 and save it. See [the documentation](http://apidock.com/rails/ActiveRecord/Persistence/increment%21).